### PR TITLE
Set-up Coding standards

### LINKED ISSUE #94: DOCUMENTING: Coding Standards
As a development team, we want to document our coding standards in our wiki to make it easier for new developers to get started.   

---

### :thought_balloon: &nbsp; Hints & Thoughts
Daniel had introduced a large part of the coding standards while redesigning the architecture and refactoring the frontend. In addition, many coding standards are based on the rules of the linter.

---

## :white_check_mark: &nbsp; Conditions of satisfaction

- [ ] There should be all our coding standards / conventions in the wiki. 
- [ ] The coding standards should be grouped by topic so that they are easy to understand.

### PR DESCRIPTION
**Pull request by SanJSp**
Fixes #94 

## Changes
In this PR a new site for our wiki regarding our coding standards is introduced. It covers our tools and some other standards we agreed on. **If you think there is something missing, please comment it!**

## Checklist before merge

**Developer's responsibilities**
* [ ] **Assign** one or two developers
* [ ] **Change code** if reviewer(s) has/have requested it
* [ ] **Pull request build** has passed
* [ ] **tested locally** (in at least chrome & firefox if frontend)
* [ ] updated the **documentation**
* [ ] added **tests** where necessary

### CODE REVIEW

File: wiki/Coding-Standards.md [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+#### Backend
+We use 
+```javascript 
+require('pathToFile')   // for imports
+exports.yourFunction()  // for exports
+```
+
+
+### Other
+- Do not fix Eslint warnings by adding the "fix-comment"
```
Comment (kej-jay) [inline wiki/Coding-Standards.md:99]: Vielleicht hier schreiben, dass wir es bestmöglich versuchen. Es ist im Ernstfall ja aber immer eine Option

**Review by kej-jay — COMMENTED** [review]
I'd like a hint to the wiki page where it says how the code should be documented. I thing that is very helpful when talking about inline comments to acknowledge other types of comments where we are actively wanting them

File: wiki/Coding-Standards.md [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+- "Single Source of Truth" is abbreviated with `Ssot` or `ssot` (don't use `SSoT` or `SSOT`)
+- use `Robot` in all cases instead of `Bot`
+- use `Application` in all cases instead of `App` (in context of supported RPA Applications)
+
+
+### General Code-Style
+- Do not use double empty lines
+- Space after openning bracket and before closing bracket (Goal: `import { Space } from AntD`)
+- Always use single-quotation marks
+- We use arrow functions only in the frontend. Please do not use the `function` keyword.
```
Comment (WolfgangDaniel) [inline wiki/Coding-Standards.md:73]: We use arrow functions everywhere not just in the frontend

File: wiki/Coding-Standards.md [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+
+**These tools enforce a lot of formatting, code-quality and simplifications that give use a base layer of standards. Please install these in your IDE and configure them as stated above.**
+
+## Coding Standards
+
+### Naming
+- Variable Names in `lowerCamelCase`
+- "Single Source of Truth" is abbreviated with `Ssot` or `ssot` (don't use `SSoT` or `SSOT`)
+- use `Robot` in all cases instead of `Bot`
+- use `Application` in all cases instead of `App` (in context of supported RPA Applications)
```
Comment (WolfgangDaniel) [inline wiki/Coding-Standards.md:66]: Here I would add:
- use PascalCase for the naming of components (.jsx files)
- use camelCase for the naming of all other files (.js files mainly)
-  use UPPERCASE for constants

**Review by WolfgangDaniel — CHANGES_REQUESTED** [review]
Looking good! Some small additions from my side.

File: wiki/Coding-Standards.md [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+
+
+### General Code-Style
+- Do not use double empty lines
+- Space after openning bracket and before closing bracket (Goal: `import { Space } from AntD`)
+- Always use single-quotation marks
+- We use arrow functions only in the frontend. Please do not use the `function` keyword.
+
+### Documentation
+Please do not use inline comments to explain the idea behind a variable or a function. Only use those for Sources where you found a special solution or woraround or for especially complex code snippets.
```
Comment (lukashueller) [inline wiki/Coding-Standards.md:76]: Little typo "workaround"
```suggestion
Please do not use inline comments to explain the idea behind a variable or a function. Only use those for Sources where you found a special solution or workaround or for especially complex code snippets.
```

**Review by lukashueller — CHANGES_REQUESTED** [review]
I think that has become a very clear document. Thanks for transferring & extending the draft to the wiki! Please fix the one little thing as well as the comments of the others and then you can merge from my point of view!

**Review by kej-jay — APPROVED** [review]
Looks good to me now

**Review by WolfgangDaniel — APPROVED** [review]
Looking good now. Thanks!

**Pull request merged**